src/classes/class-maintenancemode.php

```diff
@@ -11,78 +11,97 @@
 		 */
 		public function maintenance_mode() {
-			// Login URL for the admin
-			$login_url = wp_login_url();
-
-			// Address of the current page
-			$server_url = 'https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
-
-			// Do nothing if on admin view
-			if ( ! is_admin() ) {
-				/**
-				 * We are checking for admin role, crawler status, and important
-				 * WP pages to bypass.
-				 */
-
-				 // WP login
-				if ( false === strpos( $server_url, '/wp-login.php' ) ) {
-					return;
-				}
+			// Check option to determine the status
+			if ( get_option( 'woocart_maintenance_mode', false ) ) {
+				// Login URL for the admin
+				$login_url = wp_login_url();
+
+				// Address of the current page
+				$url = 'https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
+
+				if ( ! is_admin() ) {
+					/**
+					 * We are checking for admin role, crawler status, and important
+					 * WP pages to bypass.
+					 */
+
+					// WP login
+					if ( false !== strpos( $url, '/wp-login.php' ) ) {
+						return;
+					}
 
-				// WP admin
-				if ( false === strpos( $server_url, '/wp-admin/' ) ) {
-					return;
-				}
+					// WP admin
+					if ( false !== strpos( $url, '/wp-admin/' ) ) {
+						return;
+					}
 
-				// Media uploads
-				if ( false === strpos( $server_url, '/async-upload.php' ) ) {
-					return;
-				}
+					// Media uploads
+					if ( false !== strpos( $url, '/async-upload.php' ) ) {
+						return;
+					}
 
-				// WP upgrade file
-				if ( false === strpos( $server_url, '/upgrade.php' ) ) {
-					return;
-				}
+					// WP upgrade file
+					if ( false !== strpos( $url, '/upgrade.php' ) ) {
+						return;
+					}
 
-				// Plugins folder
-				if ( false === strpos( $server_url, '/plugins/' ) ) {
-					return;
-				}
+					// Plugins folder
+					if ( false !== strpos( $url, '/plugins/' ) ) {
+						return;
+					}
 
-				// XML-RPC
-				if ( false === strpos( $server_url, '/xmlrpc.php' ) ) {
-					return;
-				}
+					// XML-RPC
+					if ( false !== strpos( $url, '/xmlrpc.php' ) ) {
+						return;
+					}
 
-				// Custom login URL
-				if ( false === strpos( $server_url, $login_url ) ) {
-					return;
-				}
+					// Custom login URL
+					if ( false !== strpos( $url, $login_url ) ) {
+						return;
+					}
 
-				// Checking for crawlers
-				if ( $this->check_referrer() ) {
-					return;
-				}
+					// CLI
+					if ( defined( 'WP_CLI' ) && WP_CLI ) {
+						return;
+					}
 
-				// Logged-in as admin
-				if ( is_user_logged_in() ) {
-					if ( current_user_can( 'manage_options' ) ) {
+					// Checking for crawlers
+					if ( $this->check_referrer() ) {
 						return;
 					}
-				}
 
-				// Render the maintenance mode template
-				$this->render_template();
+					// Logged-in as admin
+					if ( is_user_logged_in() ) {
+						if ( current_user_can( 'manage_options' ) ) {
+							return;
+						}
+					}
+
+					// Render the maintenance mode template
+					$this->render();
```

Comment:
If we raise HTTP 418 here the default-backend will handle this as another maintenance message. This way we then have messages defined in a single component.